src/components/form/FormCompanyChallenge.vue

```diff
@@ -0,0 +1,270 @@
+<script lang="ts">
+/**
+ * FormCompanyChallenge Component
+ *
+ * @description * Use this component to render a form to create a new company
+ * challenge. This is a used on `CompanyCoordinatorPage`.
+ *
+ * @slots
+ * - `controls`: For triggering the form via dialog buttons.
+ *   exposed props and methods:
+ *     - `submit`: Method to submit the form inside the slot
+ *
+ * @components
+ * - `FormFieldDateRequired`: Component to render required date field.
+ * - `FormFieldTextRequired`: Component to render required text field.
+ *
+ * @example
+ * <form-company-challenge>
+ *
+ * @see [Figma Design](https://www.figma.com/design/L8dVREySVXxh3X12TcFDdR/Do-pr%C3%A1ce-na-kole?node-id=4858-106325&t=MqCoIBTXNV4xkXVk-1)
+ */
+
+// libraries
+import { defineComponent, ref } from 'vue';
+
+// components
+import FormFieldDateRequired from '../form/FormFieldDateRequired.vue';
+import FormFieldTextRequired from '../global/FormFieldTextRequired.vue';
+
+// types
+import type { TransportType } from '../types/Route';
+type ChallengeType = 'regularity' | 'performance';
+type ChallengeParticipants = 'individuals' | 'teams' | 'branches';
```

Comment:
Instead of branches use subsidiaries, please.

Reply:
corrected